### ISSUE TITLE
Dropdown search

### ISSUE THREAD

**Issue by SynapseScribe** [Lacerte/clima#314]
The search bar should have a fropdown list showing found location matching search string and should contain the country or country code

**Comment by prestosole** [comment]
We had considered it before, but all the choices required some sort of GMS intervention, which we don't like in Clima. However, please feel free to offer any suggestions you may have!

**Comment by triallax** [comment]
Just like @PrestoSole said, we want to make it clear that we are _not_ going to accept any GMS usage in Clima. That being said, we can use https://openweathermap.org/api/geocoding-api, which, given a _complete_ city name (plus an optional country code, and a state name for US locations, which is also optional), will give back up to five matching locations.

https://github.com/Lacerte/clima/issues/267 is related.

**Comment by SynapseScribe** [comment]
just fo reference - the open source project "forecastie" (https://f-droid.org/en/packages/cz.martykan.forecastie/) also uses openweathermap and it is able to show a list of matching locations without GMS

**Comment by triallax** [comment]
@fusionneur does it show matching locations even if you don't type the complete city name? For instance, does it show you cities named "London" if you put in "Londo"?

**Comment by prestosole** [comment]
@mhmdanas it does not, actually; It shows an "invalid city name" type error.

**Comment by SynapseScribe** [comment]
yes you're right, it seems that forecastie isn't actually able to show matches based on part of the city's name. However if there are multiple cities with same name (eg. 4 cities named Rome in US and 1 in IT) they are shown in a different list, unlike in Clima. Also it seems that both clima and forecastie are able to correct misspelled cities (eg. Valeta -> Valletta)

**Comment by triallax** [comment]
> However if there are multiple cities with same name (eg. 4 cities names Rome in US and 1 in IT) they are shown in a different list, unlike in Clima.

Well, that's exactly what I believe OpenWeatherMap supports, so we're gonna use that.

> Also it seems that both clima and forecastie are able to correct misspelled cities (eg. Valeta -> Valletta)

Clima doesn't do anything of the sort, it's almost certaintly OpenWeatherMap doing that.
